internal/supervisor/installation.go

```diff
@@ -483,6 +522,60 @@ func (s *InstallationSupervisor) createClusterInstallation(cluster *model.Cluste
 	return clusterInstallation
 }
 
+// installationCanBeScheduledOnCluster checks if the given installation can be
+// scheduled on the given cluster in regards to configuration and state. This
+// does not include resource checks.
+func (s *InstallationSupervisor) installationCanBeScheduledOnCluster(cluster *model.Cluster, installation *model.Installation, logger log.FieldLogger) bool {
+	if cluster.State != model.ClusterStateStable {
+		logger.Debugf("Cluster %s is not stable (currently %s)", cluster.ID, cluster.State)
+		return false
+	}
+	if !cluster.AllowInstallations {
+		logger.Debugf("Cluster %s is set to not allow for new installation scheduling", cluster.ID)
+		return false
+	}
+
+	existingClusterInstallations, err := s.store.GetClusterInstallations(&model.ClusterInstallationFilter{
+		PerPage:   model.AllPerPage,
+		ClusterID: cluster.ID,
+	})
+	if err != nil {
+		logger.WithError(err).Error("Failed to get existing cluster installations")
+		return false
+	}
+
+	////////////////////////////////////////////////////////////////////////////
+	//                              MULTI-TENANCY                             //
+	////////////////////////////////////////////////////////////////////////////
+	// Current model:                                                         //
+	// - isolation=true  | 1 cluster installations                            //
+	// - isolation=false | X cluster installations, where "X" is as many as   //
+	//                     will fit with the given CPU and Memory threshold.  //
+	////////////////////////////////////////////////////////////////////////////
+	if installation.Affinity == model.InstallationAffinityIsolated {
```

Comment:
You can outdent the `else` if you combine this boolean with the one below it. 0/5 on this, though, it's just a nitpick.